starlite/app.py

```diff
@@ -138,11 +137,13 @@ def create_asgi_handler(self) -> ASGIApp:
         """
         Creates an ASGIApp that wraps the ASGI router inside an exception handler.
 
-        If CORS or TruseedHost configs are provided to the constructor, they will wrap the router as well.
+        If CORS or TrustedHost configs are provided to the constructor, they will wrap the router as well.
         """
         asgi_handler: ASGIApp = self.asgi_router
-        if self.gzip_config:
-            asgi_handler = GZipMiddleware(app=asgi_handler, **self.gzip_config.dict())
+        if self.compression_config and self.compression_config.backend:
```

Comment:
Why would the `and` be required here? If it's. Pydantic model you should already have validation.